hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2241,7 +2377,7 @@ public static HoodieData<HoodieRecord> readSecondaryKeysFromBaseFiles(HoodieEngi
       } else {
         readerSchema = tableSchema;
       }
-      return createSecondaryIndexGenerator(metaClient, engineType, logFilePaths, readerSchema, partition, dataFilePath, indexDefinition,
+      return createSecondaryIndexGenerator(metaClient, engineType, new ArrayList<>(logFilePaths), readerSchema, partition, dataFilePath, indexDefinition,
```

Comment:
not sure if we need this fix if we not getting rid of entries to be deleted from SI processed separately. 

I am talking about 
```
 // Build a list of keys that need to be removed. A 'delete' record will be emitted into the respective FileGroup of
    // the secondary index partition for each of these keys.
    HoodieData<String> keysToRemove = getRecordKeysDeletedOrUpdated(engineContext, commitMetadata, dataWriteConfig.getMetadataConfig(), dataMetaClient, instantTime);

    HoodieIndexDefinition indexDefinition = getIndexDefinition(indexPartition);
    // Fetch the secondary keys that each of the record keys ('keysToRemove') maps to
    HoodiePairData<String, String> recordKeySecondaryKeyMap =
        metadata.getSecondaryKeys(keysToRemove, indexDefinition.getIndexName(), dataWriteConfig.getMetadataConfig().getSecondaryIndexParallelism());
    HoodieData<HoodieRecord> deleteRecords = recordKeySecondaryKeyMap.map(
        (recordKeyAndSecondaryKey) -> HoodieMetadataPayload.createSecondaryIndexRecord(recordKeyAndSecondaryKey.getKey(), recordKeyAndSecondaryKey.getValue(), indexDefinition.getIndexName(), true));
```

In HoodieBackedTableMetadataWriter.getSecondaryIndexUpdates

lets sync up

Reply:
Actually this is a remnant of one of the previous commits. you're right we may not need this. I'll check and remove if not needed.